src/main/java/roomescape/application/PaymentService.java

```diff
@@ -0,0 +1,62 @@
+package roomescape.application;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import roomescape.presentation.dto.request.PaymentProcessRequest;
+import roomescape.application.exception.PaymentException;
+import roomescape.infrastructure.thirdparty.PaymentRestClient;
+import roomescape.domain.Payment;
+import roomescape.infrastructure.repository.PaymentRepository;
+
+@Service
+public class PaymentService {
+
+    private final PaymentRestClient paymentRestClient;
+    private final ObjectMapper objectMapper;
+    private final PaymentRepository paymentRepository;
+
+    public PaymentService(PaymentRestClient paymentRestClient,
+                          ObjectMapper objectMapper,
+                          PaymentRepository paymentRepository
+    ) {
+        this.paymentRestClient = paymentRestClient;
+        this.objectMapper = objectMapper;
+        this.paymentRepository = paymentRepository;
+    }
+
+    public Payment process(PaymentProcessRequest request) {
+        ResponseEntity<String> paymentResponse = paymentRestClient.getPaymentResponse(request);
+        String body = paymentResponse.getBody();
+
+        JsonNode jsonNode = getJsonNode(body);
+
+        if (paymentResponse.getStatusCode() == HttpStatus.OK) {
+            String paymentKey = jsonNode.get("paymentKey").asText();
+            String orderId = jsonNode.get("orderId").asText();
+            Payment payment = Payment.create(paymentKey, orderId);
+            return paymentRepository.save(payment);
+        }
+
+        String message = jsonNode.get("message").asText();
+
+        if (paymentResponse.getStatusCode() == HttpStatusCode.valueOf(401) ||
```

Comment:
```suggestion
        if (paymentResponse.getStatusCode() == HttpStatus.UNAUTHORIZED ||
```
상태코드보단 `HttpStatus.UNAUTHORIZED`를 직접 사용하는 것이 가독성과 명확성 측면에서 좋습니다.
HttpStatus가 HttpStatusCode의 구현체라서 비교가 가능해요.
isSameCodeAs()를 사용해도 좋겠네요~

Reply:
웬만하면 Enum을 사용하려고 노력했는데 빠뜨린 부분이 있었네요!
체크해주셔서 감사합니다 라라~!
덕분에 `isSameCodeAs()` 메서드도 사용해볼 수 있었어요!
86ec7062 에 반영했습니다.